cpp/include/cudf/copying.hpp

```diff
@@ -114,6 +114,79 @@ std::unique_ptr<column> allocate_like(column_view input, size_type size,
  */
 std::unique_ptr<table> empty_like(table_view input_table);
 
+/**
+ * @brief Copies a range of elements in-place from one column to another.
+ *
+ * Overwrites the range of elements in @p target indicated by the indices
+ * [@p target_begin, @p target_begin + N) with the elements from @p source
+ * indicated by the indices [@p source_begin, @p source_end) (where N =
+ * (@p source_end - @p source_begin)). Use the out-of-place copy function
+ * returning std::unique_ptr<column> for uses cases requiring memory
+ * reallocation. For example for strings columns and other variable-width types.
+ *
+ * If @p source and @p target refer to the same elements and the ranges overlap,
+ * the behavior is undefined.
+ *
+ * @throws `cudf::logic_error` if memory reallocation is required (e.g. for
+ * variable width types).
+ * @throws `cudf::logic_error` for invalid range (if
+ * @p source_begin > @p source_end, @p source_begin < 0,
+ * @p source_begin >= @p source.size(), @p source_end > @p source.size(),
```

Comment:
@jrhemstad @harrism How do you guys think about this? This behavior might be surprising for some users.

Reply:
> wrap around to index - source.size() when index > source.size()

What is `index`?

This sounds like it will increase complexity without an explicit or strong need to do so, so I don't  think it makes sense to do so right now. `cudf::copy_range` is modeled off of `std::copy`, and what you proposed would a pretty significant deviation from that behavior.

Reply:
I agree it's not worth much effort right now. I was only suggesting we consider it if it was trivial.